### DISCUSSION TITLE
Including an overall Charter for AI Safety that is reflected and amplified by all our connected projects.

### DISCUSSION THREAD

**Discussion by fcilli** [OWASP/www-project-ai-security-and-privacy-guide#63] [Ideas]
### A case for including a Safety rule of thumb in the form of a score vector

We know that current "AI" implementations are an evolving conglomerate of decades old concepts and research, that are culminating in a new form of software dynamics. For as much as it seems far in the future, the acceleration of all things digital is rushing towards us like flies on our windscreen.

For this very reason I simply ask that in our AI Exchange initiative, and not just in LLM or ML or LowCode, we start discussing on how we  could evaluate the safety of a truly automated and malleable software based system. 

In the recent years we have seen the canonical backend and full stack software become more and more portable, thanks to containerization. We had self running isos on USBs, first, then software defined virtualized OSs, and then containers and self replicating jailed environments that thanks to the cloud can continue to work despite a full failure of a computing region.

Having the current AI tools converge with this tendency, will soon bring a completely fluid software stack, whose coherence is only dictated by its own structure boundaries. With the algorithm and data converging in a unique blend of information, this is probably the first time in computing history when we can truly say that we'll need to care about information privacy and security, more than software and data's.

To make AI systems gradually more aware, we will no longer rely on how they treat "data" but on how much they understand the information that data carries in that context. We're definitely getting there.

The idea I would love to discuss is: why don't we form a forward looking team that tries to correlate the signs of this and responds by mitigating a potential new form of security, privacy and in one word safety, that couldn't be limited by current tools?

[The CASE for a core assessment tool that helps Developers, Business and CISOs down to the Users safely Embrace AI - v.1.0a (1).pdf](https://github.com/user-attachments/files/16011018/The.CASE.for.a.core.assessment.tool.that.helps.Developers.Business.and.CISOs.down.to.the.Users.safely.Embrace.AI.-.v.1.0a.1.pdf)

**Comment by robvanderveer** [comment]
Dear fcilli. I commend your drive for AI safety. The AI Exchange is also driven, when it comes to AI security, mostly because it's such an important part of AI safety. But, as you may know, AI safety is much more than that. The other aspects are for example non-adversarial model quality, bias, the use case,  and transparency. My team wants to do its part, and that is security.
In my keynote in Lisbon (see https://www.softwareimprovementgroup.com/wp-content/uploads/20240628-RobvanderVeer-OWASPKeynote-handout-v8_compressed.pdf) I explain that security folks first need to get security right, and then we can go down the rabbit holes of all the other safety-related aspects of AI. As a team we need to focus.
An initiative connected to the Exchange, closest to what you describe is the work that some of our experts do for the AI institute safety consortium. To Learn more about that, I welcome you to contact John Sotiropoulos. If that doesn't work out ,let me know. 
Godspeed.
